proto/noble/dollar/vaults/v2/tx.proto

```diff
@@ -0,0 +1,688 @@
+    (gogoproto.nullable) = false,
+    (amino.dont_omitempty) = true
+  ];
+
+  // Yield portion of withdrawal
+  string yield_portion = 3 [
+    (cosmos_proto.scalar) = "cosmos.Int",
+    (gogoproto.customtype) = "cosmossdk.io/math.Int",
+    (gogoproto.nullable) = false,
+    (amino.dont_omitempty) = true
+  ];
+
+  // Estimated time the request becomes claimable
+  google.protobuf.Timestamp expected_claimable_at = 4 [
+    (gogoproto.nullable) = false,
+    (gogoproto.stdtime) = true
+  ];
+}
+
+// MsgProcessWithdrawalQueue allows authority to process withdrawal queue
+message MsgProcessWithdrawalQueue {
+  option (cosmos.msg.v1.signer) = "authority";
+  option (amino.name) = "dollar/vaults/v2/ProcessWithdrawalQueue";
+
+  option (gogoproto.equal) = false;
+  option (gogoproto.goproto_getters) = false;
+
+  // Authority processing the queue
+  string authority = 1 [(cosmos_proto.scalar) = "cosmos.AddressString"];
+
+  // Maximum number of requests to process
+  int32 max_requests = 2;
+}
+
+// MsgProcessWithdrawalQueueResponse returns processing results
+message MsgProcessWithdrawalQueueResponse {
+  // Number of requests processed
+  int32 requests_processed = 1;
+
+  // Total amount processed
+  string total_amount_processed = 2 [
+    (cosmos_proto.scalar) = "cosmos.Int",
+    (gogoproto.customtype) = "cosmossdk.io/math.Int",
+    (gogoproto.nullable) = false,
+    (amino.dont_omitempty) = true
+  ];
+
+  // Total amount distributed
+  string total_amount_distributed = 3 [
+    (cosmos_proto.scalar) = "cosmos.Int",
+    (gogoproto.customtype) = "cosmossdk.io/math.Int",
+    (gogoproto.nullable) = false,
+    (amino.dont_omitempty) = true
+  ];
+
+  // Remaining requests in queue
+  int32 remaining_requests = 4;
+}
+
+// ========== Spec-aligned Managed Vault Operations ==========
+
+// MsgCreateRemotePosition deploys capital to a remote ERC-4626 position
+message MsgCreateRemotePosition {
+  option (cosmos.msg.v1.signer) = "manager";
+  option (amino.name) = "dollar/vaults/v2/CreateRemotePosition";
+
+  option (gogoproto.equal) = false;
+  option (gogoproto.goproto_getters) = false;
+
+  // Authorized manager address
+  string manager = 1 [(cosmos_proto.scalar) = "cosmos.AddressString"];
+
+  // Target ERC-4626 vault address
+  string vault_address = 2;
+
+  // Hyperlane domain ID for destination chain
+  uint32 chain_id = 3;
+
+  // Amount of $USDN to deploy
+  string amount = 4 [
+    (cosmos_proto.scalar) = "cosmos.Int",
+    (gogoproto.customtype) = "cosmossdk.io/math.Int",
+    (gogoproto.nullable) = false,
+    (amino.dont_omitempty) = true
+  ];
+
+  // Minimum acceptable Noble vault shares to receive (slippage protection)
+  string min_shares_out = 5 [
+    (cosmos_proto.scalar) = "cosmos.Int",
+    (gogoproto.customtype) = "cosmossdk.io/math.Int",
+    (gogoproto.nullable) = false,
+    (amino.dont_omitempty) = true
+  ];
+}
+
+message MsgCreateRemotePositionResponse {
+  // Assigned position ID
+  string position_id = 1;
```

Comment:
In the [specs](https://github.com/noble-assets/dollar/pull/56/files#diff-dd437411e5e4b0775b33bbebed5a4657f348bcedf4cf51af27f905ec2dca5671R48) the `position_id` is defined as a `uint64`, which one is the correct type? 

Also, how do we generate it if we use a string, is it just an incremental number casted as a string or a combination of values like `{address}-{address_incremental_id}`?